external/storm-hdfs/README.md

```diff
@@ -240,6 +240,23 @@ If you are using Trident and sequence files you can do something like this:
                 .addRotationAction(new MoveFileAction().withDestination("/dest2/"));
 ```
 
+### Data Partitioning
+Data can be partitioned to different HDFS directories based on characteristics of the tuple being processed or purely
+external factors, such as system time.  To partition your your data, write a class that implements the ```Partitioner```
+interface and pass it to the withPartitioner() method of your bolt. The getPartitionPath() method returns a partition 
+path for a given tuple.
+
+Here's an example of a Partitioner that operates on a specific field of data:
+
+```java
+
+    Partitioner partitoner = new Partitioner() {
+            @Override
+            public String getPartitionPath(Tuple tuple) {
+                return Path.SEPARATOR + "city=" + tuple.getStringByField("city");
```

Comment:
I thought about having Partitioner returning an actual path but decided against it for two reasons:
- I liked the idea of the "partition" being solely a function of the tuple without reference to anything else
- Since end users implement a Partitioner having it return a complete path would give the user access to details otherwise hidden from their code.